src/components/app.tsx

```diff
@@ -119,25 +118,23 @@ export class App extends React.Component<AppProps, AppState> {
             statusMessage:undefined,
             secondGraph:false
         };
-        
+
         this.messages = SensorStrings.messages as IStringMap;
         this.connectCodap = this.connectCodap.bind(this);
+
         this.onSensorConnect = this.onSensorConnect.bind(this);
         this.onSensorData = this.onSensorData.bind(this);
+        this.onSensorStatus = this.onSensorStatus.bind(this);
         this.onSensorCollectionStopped = this.onSensorCollectionStopped.bind(this);
-        this.onSensorDisconnect = this.onSensorDisconnect.bind(this);
-        
+
         setTimeout(this.connectCodap, 1000);
-        this.sensorConnector = new SensorConnectorInterface();
-        this.sensorConnector.on("interfaceConnected", this.onSensorConnect);
-        this.sensorConnector.on("interfaceRemoved", this.onSensorConnect);
-        this.sensorConnector.on("columnAdded", this.onSensorConnect);
-        this.sensorConnector.on("columnMoved", this.onSensorConnect);
-        this.sensorConnector.on("columnRemoved", this.onSensorConnect);
-        this.sensorConnector.on("datasetAdded", this.onSensorConnect);
-        this.sensorConnector.on("data", this.onSensorData);
-        this.sensorConnector.startPolling(SENSOR_IP);
-        
+
+        let listeners = this.props.sensorManager.listeners;
+        listeners.onSensorConnect = this.onSensorConnect;
+        listeners.onSensorData = this.onSensorData;
+        listeners.onSensorStatus = this.onSensorStatus;
```

Comment:
@kswenson do you have a preference between the 1st and 3rd observer pattern options described here:   https://github.com/millermedeiros/js-signals/wiki/Comparison-between-different-Observer-Pattern-implementations 
The 2nd option doesn't seem good because I'm not going to change everything in the project to use a single dispatcher. I like the type checking of the 3rd option, but it is not standard.

Reply:
I was imagining something more like the first option, in part because it seems like a less radical change from what we have. For instance, the client could call:
```
sensorManager.addListener('onSensorConnect', this.onSensorConnect);
```
and later:
```
sensorManager.removeListener('onSensorConnect', this.onSensorConnect);
```
and the SensorManager itself would have to change relatively little.